text/0477-blueprints-update.md

```diff
@@ -0,0 +1,248 @@
+- Start Date: 2019-04-17
+- Relevant Team(s): Ember CLI
+- RFC PR: https://github.com/emberjs/rfcs/pull/477
+- Tracking: (leave this empty)
+
+# Syncing Blueprints
+
+## Summary
+
+I want to use [ember-cli-update](https://github.com/ember-cli/ember-cli-update) to keep any blueprint up-to-date. Since ember-cli-update can not possibly contain knowledge of all blueprints, some state must be stored in a project.
+
+## Motivation
+
+Projects like [libkit](https://github.com/tildeio/libkit), [@glimmer/blueprint](https://github.com/glimmerjs/glimmer.js/tree/master/packages/%40glimmer/blueprint), [@ember/octane-app-blueprint](https://github.com/ember-cli/ember-octane-blueprint), and other app replacement blueprints become out-of-date instantly after generation. Tweaks to the blueprints are only beneficial to newly generated apps.
+
+## Detailed design
+
+**State storage schema**
+
+Up until now, ember-cli-update could update apps without any configuration, but in order to handle all blueprints and their options, some state has to be stored in the project.
+
+The code is already completed https://github.com/ember-cli/ember-cli-update/pull/552, but I wanted to publish this RFC before releasing it into the wild. I want to make sure the format of these files are agreed upon, because if this is successful, it will probably propagate throughout the ecosystem.
+
+Where should the state go?
+
+```js
+// .ember-cli
+{
+  "ember-cli-update": {
+    "blueprints": [
+      {
+        "name": "my-custom-blueprint",
+        "version": "0.0.1"
+      }
+    ]
+  }
+}
+```
+
+```js
+// ember-cli-update.json
+{
+  "blueprints": [
+    {
+      "name": "my-custom-blueprint",
+      "version": "0.0.1"
+    }
+  ]
+}
+```
+
+```js
+// package.json
+{
+  // ...
+  "ember-cli-update": {
+    "blueprints": [
+      {
+        "name": "my-custom-blueprint",
+        "version": "0.0.1"
+      }
+    ]
+  }
+}
+```
+
+I vote “ember-cli-update.json”. Since the code will be modified by the updater, it’s easier to reformat JSON. “.ember-cli” I think supports JS, so it would be hard to modify. “package.json” seems like an abuse to put more metadata in there.
```

Comment:
Agreed, lets drop the other examples and focus specifically on the stand alone `ember-cli-update.json`.